snake.py

```diff
@@ -0,0 +1,227 @@
+import pygame, sys
+from pygame import *
+import random
+
+class Player(object):
+    '''
+    Player's snake object handling head position and direction, body part position and order, and moving and drawing its parts.
+    '''
+
+    def __init__(self,pos,direction):
+
+        self.pos = pos
+        self.direction = direction
+        self.size = 20
+        self.body = []
+
+        ### Colors
+        PURPLE = (128,0,128)
+        DARK_BLUE = (0,0,204)
+        BLUE = (128,128,255)
+
+        self.head_color = DARK_BLUE
+        self.body_color = BLUE
+
+
+    def __repr__(self):
+
+        return 'Head position: '+str(self.pos)+' direction: '+str(self.direction)
+
+    def step(self):
+        '''
+        Advances the player's snake one move forward.
+        '''
+        if len(self.body)>0:
+            self.body.pop(0)
+            self.body.append(self.pos)
+        direction = self.direction
+        size = self.size
+        self.pos = (self.pos[0]+direction[0]*size,self.pos[1]+direction[1]*size)
+
+    def draw(self,surface):
+        '''
+        Draws the player's snake on the surface given.
+        '''
+        for n in self.body:
+            pygame.draw.circle(surface,self.body_color,n,int(self.size/2))
+        pygame.draw.circle(surface,self.head_color,self.pos,int(self.size/2))
+
+    def grow(self):
+        '''
+        Extends the length of the body by adding an element at the current head position.
+        '''
+        self.body.append(self.pos)
+
+def final_screen():
+    """ Function that prints GAME OVER text
+    """
+    global p1
```

Comment:
Is there a particular reason why these two players needed to be declared as global variables? Could you avoid using global variables?